social/app/models/node.py

```diff
@@ -105,8 +141,46 @@ def get_public_posts(self, page=1, size=50, next_url=None):
                 % url)
             return []
 
-    def create_or_update_remote_author(self, uuid):
-        response = self._get_author(uuid)
+    def create_or_update_remote_post(self, post_uuid):
+        response = self._get_post(post_uuid)
+
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            if response.status_code == requests.codes.not_found:
+                # Post not found
+                return None
+            else:
+                raise
+
+        json = response.json()
+        posts_json = json["posts"]
+        for post_json in posts_json:
+            if post_json["id"] == post_uuid:
+                from social.app.models.post import Post
+                author_json = post_json['author']
+                from social.app.models.author import Author
+                author_id = Author.get_id_from_uri(author_json['id'])
+                author = Author.objects.get(id=author_id)
+                post, created = Post.objects.update_or_create(
+                    id=uuid.UUID(post_json["id"]),
+                    defaults={
+                        'title': post_json['title'],
+                        'description': post_json['description'],
+                        'author': author,
+                        'published': post_json['published'],
+                        'content': post_json['content'],
+                        'visibility': post_json['visibility'],
+                    }
+                )
+
+                comments = post_json['comments']
+                count = post_json["count"]
+                return post, comments, count
+        return None
```

Comment:
The `except`ional case may return this `None` and could log a warning for debugging purposes. I'm assuming you will want to add a `try`-`except` to protect against bad or unexpected response data from remote nodes.